docs/applications/big-data/big-data-in-the-linode-cloud-streaming-data-processing-with-apache-storm.md

```diff
@@ -0,0 +1,1095 @@
+
+        This value can either be the datacenter's ID or location or abbreviation. To see a list of all datacenters:
+
+            ./zookeeper-cluster-linode.sh datacenters api_env_linode.conf
+
+    -  `IMAGE_ROOT_PASSWORD` - **REQUIRED**
+
+        The default root user password for the image. All nodes of any clusters created from this image will have this as the root password, unless it's overridden in a cluster's configuration file.
+
+    -  `IMAGE_ROOT_SSH_PUBLIC_KEY` and `IMAGE_ROOT_SSH_PRIVATE_KEY`
+
+        The keypair files for SSH public key authentication as root user. Any user who logs in with this private key will be authenticated as root.
+
+        By default, the `cluster_manager.sh` setup has already created a keypair named `clusterroot` and `clusterroot.pub` under `~/.ssh/`. If you wish to replace these with your own keypair, you may create your own keys and set their full paths here.
+
+    -  `IMAGE_DISABLE_SSH_PASSWORD_AUTHENTICATION`
+
+        This disables SSH password authentication and allows only key based SSH authentication for the cluster nodes. Password authentication is considered less secure, and is hence disabled by default. To enable password authentication, you can change this value to `no`.
+
+    -  `IMAGE_ADMIN_USER`
+
+        Administrators or developers may have to log in to the cluster nodes for maintenance. Instead of logging in as root users, it's better to log in as a privileged non-root user. The script creates a privileged user with this name in the image (and in all cluster nodes based on this image).
+
+    -  `IMAGE_ADMIN_PASSWORD` - **REQUIRED**
+
+        Sets the password for the `IMAGE_ADMIN_USER`.
+
+    -  `IMAGE_ADMIN_SSH_AUTHORIZED_KEYS`
+
+        A file that contains public keys of all personnel authorized to log in to cluster nodes as `IMAGE_ADMIN_USER`. This file should be in the same format as the standard SSH *authorized_keys* file. All the entries in this file are appended to the image's `authorized_keys` file, and get inherited into all nodes based on this image.
+
+        By default, the `cluster_manager.sh` setup creates a new `clusteradmin` keypair, and this variable is set to the path of the public key. You can either retain this generated keypair and distribute the generated private key file `~/.ssh/clusteradmin` to authorized personnel. Alternatively, you can collect public keys of authorized personnel and append them to `~/.ssh/clusteradmin.pub`.
+
+    -  `IMAGE_DISK_SIZE`
+
+        The size of the image disk in MB. The default value of 5000MB is generally sufficient, since the installation only consists of the OS with Java and Zookeeper software installed.
```

Comment:
@alexfornuto  There were a couple of reasons for this decision:

+ Most important reason was faster cloning. Since an image is cloned for every node in cluster, I noticed cluster creation is relatively faster if disk sizes are small. A related observation was that cloning in general was slow when image was created in a different datacenter from the one where clones are created. Small disk sizes help in that scenario too. The software allows creating image in one DC and creating clusters in another.

+ Linode allocates 10 GB maximum space for all images of an account. Experimentally I've seen it go above that upto around 12 GB before image creation failed. Although 5000 MB disk image doesn't result in 5000 MB consumption of total space for images, I thought it prudent to default to small disk sizes to avoid image creation failures due to large disks and to enable users to create many multiple images if needed.

+ It provides flexibility to users, for example, if they prefer to store data in separate disks on the cluster nodes, separated from the OS and software. This is the recommended Zookeeper conf for performance reasons - keeping its data on a separate disk so that OS or other software writes don't hamper Zookeeper's write queues.